core/object/object.cpp

```diff
@@ -733,12 +733,24 @@ void Object::setvar(const Variant &p_key, const Variant &p_value, bool *r_valid)
 }
 
 Variant Object::callv(const StringName &p_method, const Array &p_args) {
+	int argcount = p_args.size();
 	const Variant **argptrs = nullptr;
+	Variant *args = nullptr;
 
-	if (p_args.size() > 0) {
-		argptrs = (const Variant **)alloca(sizeof(Variant *) * p_args.size());
-		for (int i = 0; i < p_args.size(); i++) {
-			argptrs[i] = &p_args[i];
+	if (argcount > 0) {
+		argptrs = (const Variant **)alloca(sizeof(Variant *) * argcount);
+
+		// Make copies of the arguments if the array is read-only, otherwise the address of p_arguments->read_only will be copied for each
```

Comment:
```suggestion
		// Make copies of the arguments if the array is read-only, otherwise the address of p_arguments->read_only will be copied for each.
```